src/Transform/ONNX/Rewrite.td

```diff
@@ -100,4 +100,56 @@ def ConvOpPaddingPattern: Pat<
   [(HasNonZeroInArrayAttr:$pads), (IsNotStringAttrOfValue<"VALID"> $auto_pad)]
 >;
 
+//===----------------------------------------------------------------------===//
+// ONNXScalerOp %X, %Offest, %Scale
+// x input, a offset, b scale
+//===----------------------------------------------------------------------===//
+// Useful test definitions.
+def AttributeIsNull :
+    Constraint<CPred<"! ($_self)">,
+  "Attribute is null">;
+
+def HasFloatType : Constraint<CPred<"(($_self).getType().dyn_cast<ShapedType>().getElementType().isF32())">>;
+  
+// Create a DenseElementsAttr from an ArrayAttr.
+def createDenseArrayAttr:
+  NativeCodeCall<"createDenseArrayAttr($_builder, $0)">;
+
+def ScalerT : NativeCodeCall<"$_builder.getI64IntegerAttr(0)">;
+
+// No attribute
+def ScalerNullPattern : Pat<
+  (ONNXScalerOp $x, $a, $b),
+  (replaceWithValue $x),
+    [(HasFloatType:$x),(AttributeIsNull:$a), (AttributeIsNull:$b)]>;
+
+// No attribute, input x not float type
+def ScalerNullPattern2 : Pat<
+  (ONNXScalerOp $x, $a, $b),
+  (ONNXCastOp $x, (ScalerT)),
```

Comment:
How did you fix the compilation error for builder of CastOp?

Reply:
The compilation error is still there where when I have other functions (ONNXSubOp, ONNXMulOp) use the output of ONNXCastOp. That is why I only have cast for ScalerOp with no attributes. 
General Scaler Op is (Mul scale, (Sub x, offset)) and Cast should be applied for all x that is not float. If you apply the current patterns to an int type input tensor, you will see that the output of Sub function is tensor of int even though offset is float, which I do not think has correct value.